lib/fenrir_view/component/accessibility_screenshot.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'mini_magick'
+
+module FenrirView
+  class Component
+    class AccessibilityScreenshot
+      attr_reader :screenshot
+
+      delegate :data, :height, :width, to: :screenshot
+
+      def initialize(screenshot:)
+        @screenshot = screenshot
+      end
+
+      def cropped(width:, height:, left:, top:)
+        image = MiniMagick::Image.open(tempfile.path)
+        image.crop("#{width}x#{height}+#{left}+#{top}")
+
+        "data:#{data_parts[1]};base64,#{Base64.strict_encode64(image.to_blob)}"
+      end
+
+      private
+
+      def data_parts
+        @data_parts ||= data.match(%r{\Adata:([-\w]+/[-\w+.]+)?;base64,(.*)}m) || []
```

Comment:
I'm surprised we have to match. Is this more performant or somehow better than using MiniMagick to get the info we need for data_parts[1]?

My assumption would be that we could move `MiniMagick::Image.open(tempfile.path)` to a private image method and then use `image.type` or `image.mime_type`.

Reply:
We need the extension to create the tempfile and, annoyingly, if we don't set the extension there then MiniMagick throws a wobbler as it doesn't know how to treat the image (even though I guess it could probably work it out).

Reply:
Oh I see, we can leave it as is then 👍